Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -187,7 +193,7 @@ IRelocator relocator
         newDocument = newDocument with { LastChange = lastChange };
         if (newDocument.SymbolTable.Resolved) {
           var resolvedDocument = newDocument with {
-            ImplementationIdToView = migratedImplementationViews,
+            ImplementationIdToView = new (migratedImplementationViews),
```

Comment:
?

Reply:
I think I understand this part better now, but I'm surprised that this sort of cloning doesn't need a lock (to make sure that the stuff that's getting cloned is all snapshot together

Reply:
This part has been changed now.